spring-boot-project/spring-boot-actuator-autoconfigure/src/main/java/org/springframework/boot/actuate/autoconfigure/tracing/otlp/OtlpTracingConfigurations.java

```diff
@@ -76,15 +80,55 @@ static class Exporters {
 		OtlpHttpSpanExporter otlpHttpSpanExporter(OtlpProperties properties,
 				OtlpTracingConnectionDetails connectionDetails) {
 			OtlpHttpSpanExporterBuilder builder = OtlpHttpSpanExporter.builder()
-				.setEndpoint(connectionDetails.getUrl())
+				.setEndpoint(getUrl(connectionDetails))
 				.setTimeout(properties.getTimeout())
 				.setCompression(properties.getCompression().name().toLowerCase());
-			for (Entry<String, String> header : properties.getHeaders().entrySet()) {
+			for (Entry<String, String> header : getHeaders(properties).entrySet()) {
 				builder.addHeader(header.getKey(), header.getValue());
 			}
 			return builder.build();
 		}
 
 	}
 
+	private static String getUrl(OtlpTracingConnectionDetails connectionDetails) {
+		String url = connectionDetails.getUrl();
+		if (url != null) {
+			return url;
+		}
+		Map<String, String> env = System.getenv();
```

Comment:
Why don't use the existing Spring's environment ?
https://docs.spring.io/spring-framework/reference/core/beans/environment.html
> from a variety of sources: properties files, JVM system properties, system environment variables, JNDI, servlet context parameters, ad-hoc Properties objects, Map objects, and so on.